tee/test/generictest.go

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"perun.network/go-perun/pkg/test"
+
+	"github.com/perun-network/erdstall/eth"
+	"github.com/perun-network/erdstall/tee"
+)
+
+type clientAction = func(ctx context.Context, bal *tee.BalanceProof) error
+
+type Client interface {
+	Balance() *big.Int
+	AddBalance(*big.Int)
+	Address() common.Address
+	SetMinedBlockNum(uint64)
+	TxEpoch() tee.Epoch
+	UpdateLastBlockNum()
+	Deposit(context.Context, *big.Int) error
+	Exit(context.Context, *tee.BalanceProof) error
+	Withdraw(context.Context, *tee.BalanceProof) error
+	Send(common.Address, *big.Int) error
+	SendToClient(Client, *big.Int) error
+	SendInvalidTxs(*rand.Rand, common.Address) []error
+	InvalidTxs(*rand.Rand, common.Address) []*tee.Transaction
+	SignTx(*tee.Transaction)
+}
+
+type EnclaveTestConfig interface {
```

Comment:
`NewClient` won't be necessary anymore, see other comment.

Do we even need `NewTransactor`? Don't we always use the direct enclave transactor in the enclave test? If so, we can also delete `NewTransactor` and always use the `EnclaveTransactor` in the test, so we don't need `EnclaveTestConfig` at all.